[DOCS-1302] Document audit logging for SaaS

mdlinville opened the pull request:
[DOCS-1302] Document audit logging for SaaS

Previews:
- Major: https://docs-1302.docodile.pages.dev/guides/hosting/monitoring-usage/audit-logging/
- Minor: https://docs-1302.docodile.pages.dev/guides/hosting/iam/scim/

[DOCS-1302]: https://wandb.atlassian.net/browse/DOCS-1302?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

mdlinville committed:
[DOCS-1302] Document audit logging for SaaS

mdlinville committed:
[DOCS-1302] Document audit logging for SaaS

mdlinville committed:
startDate only for SaaS Cloud, fix a typo

mdlinville committed:
startDate only for SaaS Cloud, fix a typo

jennwandb commented on content/guides/hosting/monitoring-usage/audit-logging.md:
```diff
@@ -32,76 +29,118 @@ The following table lists all the different keys that might be present in your a
+|`user_asset`              | If present, action was taken on this user asset.
+|`cli_version`             | If the action is taken via python SDK, this will contain the version
+|`actor_ip`                | IP address of the logged-in user.
+|`actor_email`             | if present, action was taken on this actor email.
```
nit: This precedes the current changes, but there's a mix of capitalized/lowercase first words and ending period/no period for these Definitions. I think it would look better if these all had the same format

jennwandb commented on content/guides/hosting/monitoring-usage/audit-logging.md:
```diff
@@ -32,76 +29,118 @@ The following table lists all the different keys that might be present in your a
+
+    {{% alert %}}
+    - If you set both `startDate` and `numDays`, logs are returned from `startDate` to `startDate` + `numDays`.
+    - If you omit `startDate` but include `numDays`, logs are returned from `today` to `numDays`.fetch for `today` only.
```
nit: extra 
"fetch for `today` only." in this bullet pt

jennwandb commented on content/guides/hosting/monitoring-usage/audit-logging.md:
```diff
@@ -32,76 +29,118 @@ The following table lists all the different keys that might be present in your a
+| `run:undelete_many`           | Runs are brought back from trash in batch.
+| `run:update`                  | Run is updated.
+| `sweep:create_agent`          | Sweep agent is created.
+| `team:invite_use`r            | User is invited to team.
```
nit: `team:invite_user`

mdlinville committed:
Copyedits

- Edit and sort the schema alphabetically
- Edit and sort the actions table alphabetically
- Move the actions not collected on SaaS to a footnote
- Consolidate some admonitions
- Edit the log retention content, move it into an alert

mdlinville committed:
Copyedits

- Edit and sort the schema alphabetically
- Edit and sort the actions table alphabetically
- Move the actions not collected on SaaS to a footnote
- Consolidate some admonitions
- Edit the log retention content, move it into an alert

mdlinville committed:
Merge remote-tracking branch 'origin/main' into DOCS-1302

mdlinville committed:
Missed feedback

jennwandb commented on content/guides/hosting/monitoring-usage/audit-logging.md:
```diff
@@ -7,101 +7,136 @@ title: Track user activity with audit logs
 
-{{% alert %}}
-**Audit Log Retention:** If a compliance, security or risk team in your organization requires audit logs to be retained for a specific period of time, W&B recommends to periodically transfer the logs from your instance-level bucket to a long-term retention storage. If you're instead using the API to access the audit logs, you can implement a simple script that runs periodically (like daily or eve ...
+If you are subject to the [Health Insurance Portability and Accountability Act of 1996 (HIPAA)](https://aspe.hhs.gov/reports/health-insurance-portability-accountability-act-1996"), you must maintain audit logs must be retained for a minimum of 6 years in an environment where they cannot be deleted or modified by any internal or exterrnal actor before the end of the mandatory retention period. For ...
```
nit: `you must maintain` can be removed from `you must maintain audit logs must be retained for a minimum of 6 years in an environment...`

jennwandb commented on content/guides/hosting/monitoring-usage/audit-logging.md:
```diff
@@ -7,101 +7,136 @@ title: Track user activity with audit logs
+|`response_code`           | The HTTP response code for the action, if applicable.
+|`timestamp`               | The time of the event in [RFC3339 format](https://www.rfc-editor.org/rfc/rfc3339). For example, `2023-01-23T12:34:56Z` represents January 23, 2023 at 12:34:56 UTC.
+|`user_asset`              | The user asset the action impacts (rather than the user performing the action), if applicable.
+|`user_email`              | The email address of the user the action (rather than the email address of the user performing the action), if applicable.
```
nit: looks to be missing `impacts` from `The email address of the user the action impacts`

jennwandb commented on content/guides/hosting/monitoring-usage/audit-logging.md:
```diff
@@ -7,101 +7,136 @@ title: Track user activity with audit logs
+<a id="1">1</a>: On [SaaS Cloud]({{< relref "/guides/hosting/hosting-options/saas_cloud.md" >}}), audit logs are not collected for:
+- Open or Public projects.
+- The `report:read` action.
+- `User` actions which are not tried to a specific organization.
```
nit: `tried` -> `tied`

jennwandb reviewed (commented):
overall lgtm! added comments for little fixes

mdlinville committed:
More feedback

mdlinville committed:
Merge remote-tracking branch 'origin/DOCS-1302' into DOCS-1302

jennwandb commented on content/guides/hosting/monitoring-usage/audit-logging.md:
```diff
@@ -7,101 +7,148 @@ title: Track user activity with audit logs
 
-{{% alert %}}
-**Audit Log Retention:** If a compliance, security or risk team in your organization requires audit logs to be retained for a specific period of time, W&B recommends to periodically transfer the logs from your instance-level bucket to a long-term retention storage. If you're instead using the API to access the audit logs, you can implement a simple script that runs periodically (like daily or eve ...
+<<<<<<< HEAD
```
looks like an unresolved merge conflict

jennwandb reviewed (commented):
audit logs and scim updates lgtm other than the merge conflicts

mdlinville committed:
Merge remote-tracking branch 'origin/main' into DOCS-1302

mdlinville committed:
Clean up conflict markers

mdlinville committed:
Merge branch 'main' into DOCS-1302

mdlinville committed:
Merge branch 'main' into DOCS-1302

ngrayluna commented on content/guides/hosting/monitoring-usage/audit-logging.md:
````diff
@@ -7,101 +7,136 @@ title: Track user activity with audit logs
+To exclude PII, pass the `anonymize=true` URL parameter. For example, if your W&B instance URL is `https://mycompany.wandb.io` and you would like to get audit logs for user activity within the last week and exclude PII, use an API endpoint like:
+
+```text
+https://mycompany.wandb.io/admin/audit_logs?numDays=7&anonymize=true`.
````
```suggestion
https://mycompany.wandb.io/admin/audit_logs?numDays=7&anonymize=true.
```

ngrayluna reviewed (approved):
Small nit

mdlinville committed:
Update content/guides/hosting/monitoring-usage/audit-logging.md

Co-authored-by: Noah Luna <15202580+ngrayluna@users.noreply.github.com>
